### PR TITLE
Add oauth2-proxy and restrict access to apps which are using ingress #106

### LINKED ISSUE #106: OIDC Support for ingress
Support a mechanism for enforcing some ingresses to require login via OIDC first:

Try:  https://github.com/jetstack/kube-oidc-proxy or https://github.com/pusher/oauth2_proxy

### PR DESCRIPTION
**Pull request by teodor-pripoae**
Fixes #106

### CODE REVIEW

File: pkg/types/types.go [hunk 1]
```diff
@@ -160,14 +168,17 @@ type NFS struct {
...
-	AdminGroup string `yaml:"adminGroup,omitempty"`
-	BindDN     string `yaml:"dn,omitempty"`
+	Disabled         bool   `yaml:"disabled,omitempty"`
+	Host             string `yaml:"host,omitempty"`
+	Port             string `yaml:"port,omitempty"`
+	Username         string `yaml:"username,omitempty"`
+	Password         string `yaml:"password,omitempty"`
+	Domain           string `yaml:"domain,omitempty"`
+	AdminGroup       string `yaml:"adminGroup,omitempty"`
+	UserDN           string `yaml:"userDN,omitempty"`
```
Comment (moshloop) [inline pkg/types/types.go]: Can we add comments on these fields with examples so they show up in the docs?

File: manifests/nginx.yml [hunk 2]
```diff
@@ -317,3 +318,91 @@ spec:
...
+           - "--pass-authorization-header=true"
+           - "--pass-host-header=true"
+           - "--pass-access-token=true"
+           - "--set-xauthrequest=true"
+           - "--set-authorization-header=true"
+           - "--request-logging=true"
+          env:
+            - name: OAUTH2_PROXY_COOKIE_SECRET
+              value: {{.oauth2Proxy.cookieSecret}}
+          image: {{.oauth2Proxy.dockerImage}}:{{.oauth2Proxy.version}}
```
Comment (moshloop) [inline manifests/nginx.yml]:
```suggestion
          image: flanksource/oauth2_proxy:v5.0.0.flanksource.1
```

Comment (teodor-pripoae) [reply manifests/nginx.yml]: Done

File: pkg/types/types.go (hunk 1 above)
Comment (teodor-pripoae) [reply pkg/types/types.go]: Done

**Pull request merged**